src/app/modules/library/personal-library/personal-library.component.ts

```diff
@@ -1,21 +1,48 @@
-import { Component, Inject } from '@angular/core';
+import { Component, Inject, Signal, WritableSignal, computed, signal } from '@angular/core';
 import { LibraryProject } from '../libraryProject';
 import { LibraryService } from '../../../services/library.service';
 import { LibraryComponent } from '../library/library.component';
 import { MatDialogRef, MAT_DIALOG_DATA, MatDialog } from '@angular/material/dialog';
+import { ProjectFilterValues } from '../../../domain/projectFilterValues';
+import { ArchiveProjectService } from '../../../services/archive-project.service';
+import { Subscription } from 'rxjs';
+import { PageEvent } from '@angular/material/paginator';
 
 @Component({
   selector: 'app-personal-library',
   templateUrl: './personal-library.component.html',
   styleUrls: ['./personal-library.component.scss']
 })
 export class PersonalLibraryComponent extends LibraryComponent {
-  projects: LibraryProject[] = [];
   filteredProjects: LibraryProject[] = [];
-  personalProjects: LibraryProject[] = [];
-  sharedProjects: LibraryProject[] = [];
+  protected personalProjects: LibraryProject[] = [];
+  projects: LibraryProject[] = [];
+  protected selectedProjects: WritableSignal<LibraryProject[]> = signal([]);
+  protected sharedProjects: LibraryProject[] = [];
+  protected showArchived: boolean = false;
 
-  constructor(protected dialog: MatDialog, protected libraryService: LibraryService) {
+  protected numSelectedProjects: Signal<number> = computed(() => {
+    return this.selectedProjects().length;
+  });
+  protected projectIdToIsSelected: Signal<any> = computed(() => {
+    return this.selectedProjects().reduce((accumulator, project) => {
+      accumulator[project.id] = true;
+      return accumulator;
+    }, {});
+  });
+  protected selectedAllProjects: Signal<boolean> = computed(() => {
+    const numSelectedProjects = this.numSelectedProjects();
+    return numSelectedProjects !== 0 && numSelectedProjects === this.getProjectsInView().length;
+  });
+  protected selectedSomeProjects: Signal<boolean> = computed(() => {
+    return this.numSelectedProjects() !== 0 && !this.selectedAllProjects();
+  });
```

Comment:
For some of there signals, you can remove the inner bracket and the ```return``` for example:
```
  protected numSelectedProjects: Signal<number> = computed(() => 
    this.selectedProjects().length;
  );
```